src/persistenttabmodel.cpp

```diff
@@ -70,6 +68,8 @@ void PersistentTabModel::tabsAvailable(QList<Tab> tabs)
         m_loaded = true;
         emit loadedChanged();
     }
+
+    connect(this, SIGNAL(activeTabIndexChanged()), this, SLOT(saveActiveTab()));
```

Comment:
This should be unique connection as tabsAvailable will be emitted also when all tabs are closed.